include/otf2xx/definition/detail/calling_context_impl.hpp

```diff
@@ -55,22 +55,24 @@ namespace definition
 
         class calling_context_impl : public ref_counted
         {
+            using parent_ref_type = otf2::reference_impl<calling_context, calling_context>;
```

Comment:
First: should be simply `reference_type`.   
Better idea: Simplify `referable_base` to take only an `Impl` and define the tag type in the Impl. This way you can easily get a `reference` in the impl, effort is about the same `using tag_type = Foo` vs a template param and you don't need to be that careful with incomplete types. Things like this line may become out-of sync as actually `reference_impl` should only be used by `reference<>`, but now `calling_context_impl` has to "know" the tag type too.

Reply:
If I move the tag_type into the `*_impl`, I lose the type-safe `operator==` which allows to compare `referable_base` definitions with those having the same tag. While this seems strange at first, I think it is in line with the overall theme for comparison by ref.

Reply:
`referable_base` gets the impl, so it has the tag type. It then calls the `==` of the reference and you can only compare references with the same tag, right?

Reply:
Alternatively: `std::enable_if_t<std::is_same<Impl1::tag_type, Impl2::tag_type>::value>`